x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/vector/Knn.java

```diff
@@ -303,7 +349,11 @@ private static Knn readFrom(StreamInput in) throws IOException {
         Expression field = in.readNamedWriteable(Expression.class);
         Expression query = in.readNamedWriteable(Expression.class);
         QueryBuilder queryBuilder = in.readOptionalNamedWriteable(QueryBuilder.class);
-        return new Knn(source, field, query, null, null, queryBuilder);
+        List<Expression> filterExpressions = List.of();
+        if (in.getTransportVersion().onOrAfter(TransportVersions.ESQL_KNN_FUNCTION_PREFILTER)) {
```

Comment:
This may be not needed as prefilters are generated on rewriting during preplanning, so they don't need to be serialized. But, we probably will need them for the non-pushed case.

Reply:
Good point, no BWC needed for that as it's not released yet. Let me check that!

Reply:
Removed in 62b0d1cb3bc747d2e446db98928ad0f567c57b2d, thanks!